### PR TITLE
Aggregate regions

### PR DESCRIPTION
**Pull request by industrial-sloth**
This PR adds two closely-related methods to Series objects, `meanOfRegion` and `meanByRegions`. These calculate the average time series within a set or sets of Series records specified by their keys.

### CODE REVIEW

File: python/thunder/rdds/series.py [hunk 1]
```diff
@@ -561,6 +561,67 @@ def query(self, inds, var='inds', order='F', isOneBased=True):
...
+                       _MeanCombiner.mergeIntoMeanTuple,
+                       _MeanCombiner.combineMeanTuples)
+        kmean = tuple(kmean.astype('int32'))
+        return kmean, vmean
+
+    def meanByRegions(self, nestedKeys):
+        """Takes the mean of Series values within groupings specified by the passed keys.
+
+        Each sequence of keys passed specifies a "region" within which to calculate the mean. For instance,
+        series.aggregateByRegions([[1,2]]) would return the mean of the records in series with key 1 and 2.
```
Comment (freeman-lab) [inline python/thunder/rdds/series.py]: Need to update with `meanByRegions`

File: python/thunder/rdds/series.py [hunk 2]
```diff
@@ -561,6 +561,67 @@ def query(self, inds, var='inds', order='F', isOneBased=True):
...
+        bcRegionKeys = self.rdd.context.broadcast(frozenset(keys))
+        n, kmean, vmean = self.rdd.filter(lambda (k, v): k in bcRegionKeys.value) \
+            .map(lambda (k, v):  (array(k, dtype=v.dtype), v)) \
+            .aggregate(_MeanCombiner.createZeroTuple(),
+                       _MeanCombiner.mergeIntoMeanTuple,
+                       _MeanCombiner.combineMeanTuples)
+        kmean = tuple(kmean.astype('int32'))
+        return kmean, vmean
+
+    def meanByRegions(self, nestedKeys):
```
Comment (freeman-lab) [inline python/thunder/rdds/series.py]: Let's change to `meanByRegion` by analogy to `reduceByKey` (as opposed to `reduceByKeys`)

File: python/thunder/rdds/series.py (hunk 1 above)
Comment (freeman-lab) [inline python/thunder/rdds/series.py]: Maybe use tuples for the examples, e.g. `[[(1,1), (1,2)]]` and `[[(1,1), (1,2)], [(1,1), (1,3)]]`?

**Comment by industrial-sloth** [comment]
Ok @freeman-lab, have made those couple small changes. Feel free to pull this in now if you'd like - tomorrow I'll try to add the mask-based aggregation methods we've discussed, either in this PR if it's still open or a new one if not.

**Comment by freeman-lab** [comment]
Perfect, LGTM, merging it in!

**Pull request merged**